drivers/lora/sx1276.c

```diff
@@ -112,51 +109,61 @@ void BoardCriticalSectionEnd(uint32_t *mask)
 	irq_unlock(*mask);
 }
 
-uint32_t RtcGetTimerElapsedTime(void)
+u32_t RtcGetTimerValue(void)
 {
-	u32_t ticks;
-	int err;
-
-	err = counter_get_value(dev_data.counter, &ticks);
-	if (err) {
-		LOG_ERR("Failed to read counter value (err %d)", err);
-		return 0;
-	}
+	return k_ms_to_ticks_ceil32(k_uptime_get_32());
+}
 
-	return ticks;
+u32_t RtcGetTimerElapsedTime(void)
+{
+	return (k_ms_to_ticks_ceil32(k_uptime_get_32()) - saved_time);
 }
 
 u32_t RtcGetMinimumTimeout(void)
 {
-	/* TODO: Get this value from counter driver */
-	return 3;
+	return 1;
 }
 
-void RtcSetAlarm(uint32_t timeout)
+void RtcStopAlarm(void)
 {
-	struct counter_alarm_cfg alarm_cfg;
+	k_timer_stop(&dev_data.timer);
+}
 
-	alarm_cfg.flags = 0;
-	alarm_cfg.ticks = timeout;
+static void timer_callback(struct k_timer *_timer)
+{
+	ARG_UNUSED(_timer);
 
-	counter_set_channel_alarm(dev_data.counter, 0, &alarm_cfg);
+	TimerIrqHandler();
 }
 
-uint32_t RtcSetTimerContext(void)
+void RtcSetAlarm(u32_t timeout)
 {
-	return 0;
+	k_timer_start(&dev_data.timer, k_ticks_to_ms_floor32(timeout),
+		      K_NO_WAIT);
 }
 
-uint32_t RtcMs2Tick(uint32_t milliseconds)
+u32_t RtcSetTimerContext(void)
 {
-	return counter_us_to_ticks(dev_data.counter, (milliseconds / 1000));
+	saved_time = k_ms_to_ticks_ceil32(k_uptime_get_32());
+
+	return saved_time;
+}
+
+u32_t RtcGetTimerContext(void)
+{
+	return saved_time;
 }
 
-void DelayMsMcu(uint32_t ms)
+void DelayMsMcu(u32_t ms)
 {
 	k_sleep(ms);
 }
 
+u32_t RtcMs2Tick(uint32_t milliseconds)
```

Comment:
You are missing `RtcTick2Ms()` implementation.

Reply:
It was not present in initial code either so thought of adding it in LoRaWAN PR. But I can add it in this PR as well (as a separate commit though)